agent/consul/internal_endpoint.go

```diff
@@ -107,13 +107,30 @@ func (m *Internal) ServiceDump(args *structs.ServiceDumpRequest, reply *structs.
 		&args.QueryOptions,
 		&reply.QueryMeta,
 		func(ws memdb.WatchSet, state *state.Store) error {
-			index, nodes, err := state.ServiceDump(ws, args.ServiceKind, args.UseServiceKind, &args.EnterpriseMeta)
+			// Get, store, and filter nodes
+			maxIdx, nodes, err := state.ServiceDump(ws, args.ServiceKind, args.UseServiceKind, &args.EnterpriseMeta)
 			if err != nil {
 				return err
 			}
+			reply.Nodes = nodes
 
-			reply.Index, reply.Nodes = index, nodes
-			if err := m.srv.filterACL(args.Token, reply); err != nil {
+			if err := m.srv.filterACL(args.Token, reply.Nodes); err != nil {
```

Comment:
Something weird is happening with the ACL filtering, causing a test to fail? https://app.circleci.com/pipelines/github/hashicorp/consul/11574/workflows/99efb4ba-e98f-4889-919e-26d98efab9a2/jobs/216919

Reply:
Yep, it needed a pointer: https://github.com/hashicorp/consul/blob/master/agent/consul/acl.go#L1843